### PR TITLE
Release 0.14.4 backports

### PR DESCRIPTION
**Pull request by ax3l**
https://github.com/openPMD/openPMD-api/milestone/3

### Increased Compatibility & Python Install Bug

This release fixes various read/parsing bugs and increases compatibility with upcoming versions of ADIOS and old releases of Intel `icpc`. An installation issue for pip-based installs from source in the last release was fixed and Python 3.10 support added. Various documentation and installation warnings have been fixed.

Thanks to @ax3l and @franzpoeschel for contributing to this release!

### Python

```python
import openpmd_api
print(openpmd_api.__version__)
```

### C++

```C++
#include <openPMD/openPMD.hpp>
#include <iostream>

int main() {
    // query compile-time macros:
    std::cout << OPENPMDAPI_VERSION_MAJOR << "."
              << OPENPMDAPI_VERSION_MINOR << "."
              << OPENPMDAPI_VERSION_PATCH << std::endl;
    // there is also: OPENPMDAPI_VERSION_LABEL (e.g. "alpha")

    // or query the runtime API (since 0.12.0):
    std::cout << openPMD::getVersion() << std::endl;
    return 0;
}
```

### CLI

```bash
# command line option (since 0.12.0):
openpmd-ls --version
```

### CODE REVIEW

File: test/SerialIOTest.cpp [hunk 1]
```diff
@@ -4203,6 +4213,176 @@ TEST_CASE( "variableBasedParticleData", "[serial][adios2]" )
...
+        REQUIRE( !spanWorkaround );
+
+        E_compressed.storeChunk< int >(
+            { 0 }, { 10 }, [ &spanWorkaround ]( size_t size ) {
+                spanWorkaround = true;
+                return std::shared_ptr< int >(
+                    new int[ size ]{}, []( auto * ptr ) { delete[] ptr; } );
+            } );
+
+        REQUIRE( spanWorkaround );
```
Comment (ax3l) [inline test/SerialIOTest.cpp:4263]: @franzpoeschel does this test require the new JSON re-work or so? In that case, I would need to remove it from the backport.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3,7 +3,53 @@
 Changelog
 =========
 
-0.14.2
+0.14.4
+------
+**Date:** 2022-01-19
```
Comment (ax3l) [inline CHANGELOG.rst]: Memo to myself: update me once CI/test question is resolved.

File: test/SerialIOTest.cpp (hunk 1 above)
Comment (franzpoeschel) [reply test/SerialIOTest.cpp:4263]: It does, though it might actually make sense to backport the workaround too. I'll have a look

Comment (franzpoeschel) [reply test/SerialIOTest.cpp:4263]: I've opened the backport as a PR: https://github.com/ax3l/openPMD-api/pull/2

**Pull request merged**